modules/n1ql/pages/n1ql-language-reference/metafun.adoc

```diff
@@ -368,6 +368,70 @@ The return value is a flattened list of array elements for use in an array index
 
 For examples, refer to xref:n1ql:n1ql-language-reference/indexing-arrays.adoc#examples[Array Indexing Examples].
 
+[[formalize,FORMALIZE()]]
+== FORMALIZE(`statement` [ `,query_context` ])
+
+=== Description
+
+Specifies all references from a query.
+
+Can be spelled as FORMALIZE or FORMALISE.
+
+=== Arguments
+
+statement:: A string containing the statement to formalize.
+
+query_context:: [ Optional ]
+A string query context value for the function to use when formalizing.
+
+=== Return Value
+
+Returns a query with all references fully specified.
+
+=== Examples
+
+[[formalize-ex,FORMALIZE() Example]]
+====
+.Query
+[source,sqlpp]
+----
+SELECT formalize("SELECT * FROM landmark WHERE country = 'United Kingdom'","default:`travel-sample`.inventory")
+----
+.Results
+[source,json]
+----
+[
+  {
+    "$1": "select self.* from `default`:`travel-sample`.`inventory`.`landmark` where ((`landmark`.`country`) = \"United Kingdom\")"
+  }
+]
+----
+====
+
+====
+.Query
+[source,sqlpp]
+----
+SELECT statement
+      ,NVL(queryContext,"") AS queryContext
+      ,formalize(statement, queryContext) AS formalized 
+FROM system:completed_requests;
```

Comment:
Looks kind of odd having the commas at the start of the line. Reads better if they're at the end of the previous line.

```suggestion
SELECT statement,
       NVL(queryContext,"") AS queryContext,
       formalize(statement, queryContext) AS formalized 
FROM system:completed_requests;
```